ci/docker/ruster/Dockerfile

```diff
@@ -32,18 +36,33 @@ RUN set -eux; \
     chmod -R a+w $RUSTUP_HOME $CARGO_HOME; \
     rustup --version; \
     cargo --version; \
-    rustc --version; \
-    apt-get remove -y --auto-remove \
-        wget \
-        ; \
-    rm -rf /var/lib/apt/lists/*;
+    rustc --version
 
-RUN rustup target add \
-        x86_64-unknown-linux-gnu \
-        i686-unknown-linux-gnu \
-        armv7-unknown-linux-gnueabihf \
-        aarch64-unknown-linux-gnu \
-        arm-unknown-linux-gnueabi
+RUN set -eux; \
+  rustup target add \
+    x86_64-unknown-linux-gnu \
+    i686-unknown-linux-gnu \
+    armv7-unknown-linux-gnueabihf \
+    aarch64-unknown-linux-gnu \
+    arm-unknown-linux-gnueabi
+
+RUN set -eux; \
+  mkdir -p /usr/lib/i386-linux-gnu \
+    /usr/lib/x86_64-linux-gnu \
+    /usr/lib/arm-linux-gnueabihf \
+    /usr/lib/arm-linux-gnueabi \
+    /usr/lib/aarch64-linux-gnu
+
+RUN --mount=type=secret,id=gl_access_token export GL_ACCESS_TOKEN="$(cat /run/secrets/gl_access_token)"; \
+  set -eux; \
+  curl --header "PRIVATE-TOKEN: ${GL_ACCESS_TOKEN}" -o /usr/lib/i386-linux-gnu/libsqlite3.so -L ${SQLITE_DOWNLOAD_URL_PREFIX}/linux_i686_sqlite3.so; \
+  curl --header "PRIVATE-TOKEN: ${GL_ACCESS_TOKEN}" -o /usr/lib/x86_64-linux-gnu/libsqlite3.so -L ${SQLITE_DOWNLOAD_URL_PREFIX}/linux_x86_64_sqlite3.so; \
+  curl --header "PRIVATE-TOKEN: ${GL_ACCESS_TOKEN}" -o /usr/lib/arm-linux-gnueabihf/libsqlite3.so -L ${SQLITE_DOWNLOAD_URL_PREFIX}/linux_armv7_eabihf_sqlite3.so; \
+  curl --header "PRIVATE-TOKEN: ${GL_ACCESS_TOKEN}" -o /usr/lib/arm-linux-gnueabi/libsqlite3.so -L ${SQLITE_DOWNLOAD_URL_PREFIX}/linux_armv5_eabi_sqlite3.so; \
+  curl --header "PRIVATE-TOKEN: ${GL_ACCESS_TOKEN}" -o /usr/lib/aarch64-linux-gnu/libsqlite3.so -L ${SQLITE_DOWNLOAD_URL_PREFIX}/linux_aarch64_sqlite3.so; \
+  apt-get remove -y --auto-remove curl
```

Comment:
Any reason https://docs.docker.com/build/building/multi-stage/ was not used?

Reply:
As far as I remember we didn't want to do that in dynamic linking, because our docker images need cleanup in general anyway and this was going to be done with the cleanup later? Something like that, it was some time ago, not sure if I remember correctly